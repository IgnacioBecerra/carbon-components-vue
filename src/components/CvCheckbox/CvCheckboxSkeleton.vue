<template>
  <div
    class="cv-checkbox"
    :class="`${carbonPrefix}--form-item ${carbonPrefix}--checkbox-wrapper`"
  >
    <label :class="`${carbonPrefix}--checkbox-label ${carbonPrefix}--skeleton`">
      <span
        :class="`${carbonPrefix}--checkbox-label-text ${carbonPrefix}--skeleton`"
      ></span>
    </label>
  </div>
</template>

<script setup>
import { carbonPrefix } from '../../global/settings';
</script>
