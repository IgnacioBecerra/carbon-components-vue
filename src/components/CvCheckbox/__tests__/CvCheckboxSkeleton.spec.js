import { shallowMount } from '@vue/test-utils';
import { carbonPrefix } from '../../../global/settings';
import CvCheckboxSkeleton from '../CvCheckboxSkeleton.vue';

describe('CvCheckboxSkeleton', () => {
  it('renders checkbox skeleton', async () => {
    const wrapper = shallowMount(CvCheckboxSkeleton);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders checkbox skeleton with required visual elements', () => {
    const wrapper = shallowMount(CvCheckboxSkeleton);

    const skeletonWrapper = wrapper.find('.cv-checkbox');
    const skeletonLabel = skeletonWrapper.find(
      `.${carbonPrefix}--checkbox-label`
    );
    const skeletonText = skeletonWrapper.find(
      `.${carbonPrefix}--checkbox-label-text`
    );
    const expectedSkeletonClass = `${carbonPrefix}--skeleton`;

    expect(skeletonLabel.classes()).toContain(expectedSkeletonClass);
    expect(skeletonText.classes()).toContain(expectedSkeletonClass);
  });
});
