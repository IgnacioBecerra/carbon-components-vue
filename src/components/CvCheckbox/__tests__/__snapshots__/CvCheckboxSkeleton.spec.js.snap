// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvCheckboxSkeleton renders checkbox skeleton 1`] = `<div class="cv-checkbox bx--form-item bx--checkbox-wrapper"><label class="bx--checkbox-label bx--skeleton"><span class="bx--checkbox-label-text bx--skeleton"></span></label></div>`;
